libcodechecker/analyze/analyzers/analyzer_types.py

```diff
@@ -182,11 +202,46 @@ def initialize_checkers(config_handler,
 
     # Set user defined enabled or disabled checkers from the command line.
     if cmdline_checkers:
-        for checker_name, enabled in cmdline_checkers:
-            if enabled:
-                config_handler.enable_checker(checker_name)
+
+        # Construct a list of reserved checker names.
+        # (It is used to check if a profile name is valid.)
+        reserved_names = gen_name_variations(checkers)
+
+        for identifier, enabled in cmdline_checkers:
+
+            # The identifier is a profile name.
+            if identifier in context.available_profiles:
+                profile_name = identifier
+
+                if profile_name == "list":
+                    LOG.error("'list' is a reserved profile keyword. ")
+                    LOG.error("Please choose another profile name in "
+                              "{INSTALL_DIR}/config/config.json and rebuild.")
+                    sys.exit(1)
+
+                if profile_name in reserved_names:
+                    LOG.error("Profile name '" + profile_name + "' conflicts "
+                              "with a checker(-group) name.")
+                    LOG.error("Please choose another profile name in "
+                              "{INSTALL_DIR}/config/config.json and rebuild.")
```

Comment:
I think context should have the package root. Maybe this error message can be made more specific using that info. You could do the same with the error message above.